src/gowrapper/ring/ring.go

```diff
@@ -50,26 +72,34 @@ func lattigo_newRing(n uint64, moduli *C.uint64_t, moduliLen uint64) Handle14 {
 	return marshal.CrossLangObjMap.Add(unsafe.Pointer(r))
 }
 
-//export lattigo_newPoly
-func lattigo_newPoly(ringHandle Handle14) Handle14 {
-	r := getStoredRing(ringHandle)
-	return marshal.CrossLangObjMap.Add(unsafe.Pointer(r.NewPoly()))
+//export lattigo_newPolyQP
+func lattigo_newPolyQP(ringHandle Handle14) Handle14 {
+	r := getStoredRingQP(ringHandle)
+	poly := r.NewPoly()
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(&poly))
 }
 
-//export lattigo_ringAdd
-func lattigo_ringAdd(ringHandle, poly1Handle, poly2Handle, poly3Handle Handle14) {
-	r := getStoredRing(ringHandle)
-	p1 := GetStoredPoly(poly1Handle)
-	p2 := GetStoredPoly(poly2Handle)
-	p3 := GetStoredPoly(poly3Handle)
-	r.Add(p1, p2, p3)
+//export lattigo_copyNewPolyQP
+func lattigo_copyNewPolyQP(polyHandle Handle14) Handle14 {
+	srcPoly := GetStoredPolyQP(polyHandle)
+	newPoly := srcPoly.CopyNew()
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(&newPoly))
 }
 
-//export lattigo_polyCopy
-func lattigo_polyCopy(polyTargetHandle, polySrcHandle Handle14) {
-	pTarget := GetStoredPoly(polyTargetHandle)
-	pSrc := GetStoredPoly(polySrcHandle)
-	pTarget.Copy(pSrc)
+//export lattigo_ringQPAddLvl
+func lattigo_ringQPAddLvl(ringHandle Handle14, levelQ, levelP uint64, poly1Handle, poly2Handle, poly3Handle Handle14) {
```

Comment:
Worth renaming `poly3Handle` to `outputHandle` or similar as you did in the C++ bindings.

Reply:
done